### PR TITLE
Adding option to allow `open_dir` to use `fzf-lua` instead of `telescope`

### PR DESCRIPTION
**Pull request by YanniPapandreou**
## What

What does this pull request accomplish?

- [x] Adds an option `open_dir_provider` which allows the user to configure the plugin to use `fzf-lua` instead of `telescope` for the function `open_dir`. This defaults to 'telescope'.
- [x] Updates the readme to reflect this change.

## How

What code changes were made to accomplish it?

- Added a `setup` function which can be used to override a new option `open_dir_provider`. This default to 'telescope'.
- To use `fzf-lua` instead one can set `open_dir_provider = 'fzf_lua'` instead.

## Why

- [x] I want to add a new feature as mentioned in #1.

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,35 @@ If you are building a plugin that requires the utilities provided by `utils.nvim
...
+{
+    'yourname/plugin.nvim',
+    dependencies = {
+        {
+            '2kabhishek/utils.nvim',
+            opts = {
+                -- provider for results from `open_dir`
+                -- can be either 'telescope' or 'fzf_lua'
+                -- defaults to 'telescope'
+                open_dir_provider = "telescope"
```
Comment (2KAbhishek) [inline README.md]: lets change the name to `fuzzy_provider` it will give us some scope to expand this capability in future

File: lua/utils.lua [hunk 2]
```diff
@@ -112,24 +122,32 @@ end
...
+
+    if is_git_repo and M.open_dir_provider == "telescope" then
+      vim.cmd('Telescope git_files cwd=' .. dir)
+    elseif is_git_repo and M.open_dir_provider == "fzf_lua" then
+      vim.cmd('FzfLua git_files cwd=' .. dir)
+    elseif M.open_dir_provider == "telescope" then
+      vim.cmd('Telescope find_files')
+    else
+      vim.cmd('FzfLua files')
+    end
```
Comment (2KAbhishek) [inline lua/utils.lua]: might be useful to simplify this with a map and helper function, something like this (i wrote this on the web, please check syntax)

```lua
local function get_open_command(is_git_repo, dir)
  local commands = {
    git = {
      telescope = 'Telescope git_files cwd=',
      fzf_lua = 'FzfLua git_files cwd='
    },
    no_git = {
      telescope = 'Telescope find_files',
      fzf_lua = 'FzfLua files'
    }
  }

  if is_git_repo then
    return commands.git[M.fuzzy_provider] .. (dir or '')
  else
    return commands.no_git[fuzzy_provider]
  end
end

-- and then in open_dir, we could just do
vim.cmd(get_open_command(is_git_repo, dir))
```

**Comment by 2KAbhishek** [comment]
@YanniPapandreou overall this looks good, I left some comments to rename the option and update extensibility

Also I have fixed the tests on main too, we should probably rebase the branch and update the specs

Thanks again for the PR 🙏🏽

File: README.md (hunk 1 above)
Comment (YanniPapandreou) [reply README.md]: Sounds good - renamed :)

File: lua/utils.lua (hunk 2 above)
Comment (YanniPapandreou) [reply lua/utils.lua]: Makes sense - refactored to use such a map and helper function instead :)

**Comment by YanniPapandreou** [comment]
> @YanniPapandreou overall this looks good, I left some comments to rename the option and update extensibility
> 
> Also I have fixed the tests on main too, we should probably rebase the branch and update the specs
> 
> Thanks again for the PR 🙏🏽

@2KAbhishek thanks for the feedback - I've rebased and pushed some commits - I'm still getting one test failing:

```
Fail	||	utils human_time converts timestamp to human readable format	
            .../yanni/projects/nvim-dev/utils.nvim/tests/utils_spec.lua:268: Expected objects to be equal.
            Passed in:
            (string) '05 Oct 2024, 03:46 pm'
            Expected:
            (string) '05 Oct 2024, 03:46 PM'
```
Something weird is going on - but it might have to do with my lua versions - I tried changing the format string used in `os.date` in `utils.human_time`  from `'%d %b %Y, %I:%M %p'` to `'%d %b %Y, %I:%M %P'` but that didn't seem to fix it or change the result - and in any case according to a lua repl session (I have lua 5.1.5) the correct format string is with `%p` as is already coded:
```lua
> print(os.date('%d %b %Y, %I:%M %p', 100))
--01 Jan 1970, 02:01 AM
> print(os.date('%d %b %Y, %I:%M %P', 100))
--01 Jan 1970, 02:01 am
```
Are you also hitting this failure?

**Comment by 2KAbhishek** [comment]
Thanks for working on the feedback @YanniPapandreou I will be taking a look at the spec issue you mentioned

**Review by 2KAbhishek — APPROVED** [review]

**Comment by YanniPapandreou** [comment]
Thanks for merging and for all your feedback :)

**Pull request merged**